pkg/policy/api/fqdn.go

```diff
@@ -76,6 +78,21 @@ func (s *FQDNSelector) sanitize() error {
 	return err
 }
 
+// ToRegex converts the given FQDNSelector to its corresponding regular
+// expression.
+func (s *FQDNSelector) ToRegex() *regexp.Regexp {
+	var preparedMatch string
+	if s.MatchName != "" {
+		preparedMatch = strings.ToLower(dns.Fqdn(s.MatchName))
+	} else {
+		preparedMatch = matchpattern.Sanitize(s.MatchPattern)
+	}
+
+	regex, _ := matchpattern.Validate(preparedMatch)
+	return regex
```

Comment:
why are we ignoring the error? How can the user know the regex was considered valid?

Reply:
Updated function to return an error. The only caller assumes the value was already validated / sanitized, but since the function is exported it's good to return an error.